src/sentry/options/store.py

```diff
@@ -188,6 +188,8 @@ def set(self, key, value):
         If cache fails, we ignore silently since it'll get repaired later by sync_options.
         A boolean is returned to indicate if the network cache was set successfully.
         """
+        assert self.cache is not None, 'cache must be configured before mutating options'
```

Comment:
I used assertions here for consistency with the behavior in `OptionsManager` rather than introducing some other exception.